### ISSUE TITLE
Search view focus should be improved

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#116454]
1. Open search viewlet, make sure to have the replace also open
2. Press `tab` a couple of times and notice how the focus moves

I understand that this focus movement is designed behaviour since users expect the first tab from the search input to go to the replace input and I think it makes sense.
However there is a lot of very custom code to make this possible. Listening on `tab` presses [here](https://github.com/microsoft/vscode/blob/a0bbfacd8cfeeaebc1c5bd49bb7ffdfb804c4b5b/src/vs/workbench/contrib/search/browser/searchWidget.ts#L556).

This milestone I am trying to change how the `ActionBar` focus navigation works and this would include the search input inline checkbox actions - even though as an implementation they do not use an `ActionBar` they should behave consistently. I would like to make those checkboxes occupy only one tab stop and that users can use `left` and `right` to move focus between them - just like we now do in all `ActionBars`.
I tried changing this in Search however due to the custom focus handling code this proved to be a bit complex.

What I am suggestion is the following tab flow:
Search Input `->` Replace Input `->` Search inline actions `->` Replace inline Actions `->` Toggle Search Details `->` Files to Include `->` Files To Include Inline Actions `->` Files To Exclude `->` Files to Exclude inline actions
Each `->` represents a `tab` press.

Related issue for Find Widget https://github.com/microsoft/vscode/issues/116450

Let me know what you think and how you think is the best way to proceed
Thanks

fyi @sandy081 @neurrone @marcozehe

**Comment by isidorn** [comment]
Ideally we would:
1. Clean up the custom focus handling code
2. Adopt using the `ActionBar` for the FindInput. A `keybindingsEditor` is already a good example of using the `ActionBar` with inline checkboxes using `CheckboxViewItems` [here](https://github.com/microsoft/vscode/blob/6ee4ef1c2dbbe5a8763ccdce796a40be84c7ee8a/src/vs/workbench/contrib/preferences/browser/keybindingsEditor.ts#L395)

I should have time to help with this, so let me know what you think.

**Comment by roblourens** [comment]
Where else do we have behavior like this where you have to use arrow keys to move through focusable targets? Seems like everywhere else, tab always moves to the next button.

**Comment by isidorn** [comment]
@roblourens we do arrow navigation now everywhere - in all actionBars. Check out my comments in the UX channel.
I have synced with @JacksonKearl and I think we have agreed on the potential changes.

**Comment by roblourens** [comment]
I see it now after updating 👍

**Comment by isidorn** [comment]
I hope you like it 🤞

**Comment by JacksonKearl** [comment]
Going to put this on backlog until #116450 is settled, no use kicking it down the road every month.

**Comment by isidorn** [comment]
Okey. If you find time feel free to jump on that one as well since you use that widget. I think @rebornix might not have plans to look into it any time soon.

**Comment by roblourens** [comment]
It's not clear that there is anything left here that isn't part of https://github.com/microsoft/vscode/issues/116450
